### ISSUE TITLE
CSGCombiner does not update the shape when switching tabs

### ISSUE THREAD

**Issue by ellogwen** [godotengine/godot#55833]
### Godot version

v3.4.stable.arch_linux

### System information

OpenGL ES 3.0 Renderer: AMD Radeon Vega 8 Graphics (RAVEN, DRM 3.40.0, 5.10.79-1-MANJARO, LLVM 13.0.0)

### Issue description

When I place a CSGCombiner and add any CSG node type as children, the combiner does not update it's mesh after you switched tabs.

![image](https://user-images.githubusercontent.com/22261975/145687231-791daeb8-b863-4de9-a5e2-3d55eb1ef7a4.png)

This is odd, I work with csg a lot, why can I reproduce it so easily now and now notice it earlier? I will reboot to see if its a memory issue or something.

### Steps to reproduce

Create a new Project
Add a CSGCombiner to the scene
Add any CSG Node as child of that combiner
Change some settings of that child, see the mesh update
Switch tab to another 3d scene
Switch back
Change settings of the children
Mesh does not update anymore until you reparent the CSGCombiner

### Minimal reproduction project

[csg-combiner-bug.zip](https://github.com/godotengine/godot/files/7697536/csg-combiner-bug.zip)

Video:
https://youtu.be/cx8CTbt_Wi4

**Comment by ellogwen** [comment]
Nah, still the same after reboot.

**Comment by Calinou** [comment]
Only the gizmos seem to be outdated, not the CSG mesh itself.

**Comment by ellogwen** [comment]
On the video, right at the end, you can barely see that there are faint blue lines next to the rendered mesh. Just leave this here because it is barely visible. Don't know if related

**Comment by ellogwen** [comment]
> Only the gizmos seem to be outdated, not the CSG mesh itself.

Yeah you are right. wen saving the scene and reopen it, the mesh is correct. So seems to be a visual issue then, I guess

**Comment by cdemirer** [comment]
A very similar thing happens when you open a project that has 2 scenes, each having a CSG node (not necessarily CSGCombiner).
The editor will by default start with the 2nd scene tab open, which shows the CSG correctly.
When you switch to the 1st scene's tab, it'll look just like in this issue.
